src/Input/Select/Select.tsx

```diff
@@ -61,6 +55,38 @@ class Select extends React.Component<Props, State> {
     document.addEventListener('click', this.handleOnBlur, false);
   }
 
+  componentDidUpdate(prevProps: Props) {
+    const hasDifferentChildren = !isEqual(this.props.children,  prevProps.children);
+    if (hasDifferentChildren) {
+      this.setState({
+        selectedValue: this.props.value as string,
+        defaultValue: this.props.value as string,
+        floating: !!this.props.value,
+        filterValue: React.Children.map(this.props.children, data => data),
+      });
+    } else if (this.props.value !== prevProps.value) {
+      if (this.props.value && this.props.value !== this.state.defaultValue) {
+        this.setState({
+          selectedValue: this.props.value as string,
+          defaultValue: this.props.value as string,
+          floating: true,
+        });
+      } else if (this.props.value === '') {
+        this.setState({
+          selectedValue: this.props.value,
+          defaultValue: this.props.value,
+          floating: false,
+        });
+      }
+    } else if (this.props.isLoading !== prevProps.isLoading) {
```

Comment:
Think this can be simplified to remove the nested if condition
```
else if (this.props.isLoading && this.props.isLoading !== prevProps.isLoading) {
  this.setState({
    isLoading: this.props.isLoading,
  });
}
```

Reply:
agreed, updated